sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/HDFSBackedStateStoreProvider.scala

```diff
@@ -282,8 +282,12 @@ private[state] class HDFSBackedStateStoreProvider(
       // target file will break speculation, skipping the rename step is the only choice. It's still
       // semantically correct because Structured Streaming requires rerunning a batch should
       // generate the same output. (SPARK-19677)
+      // Also, a tmp file of delta file that generated by the first batch after restart
```

Comment:
This comment is not 100% correct, this may also happen in a speculation task.

This PR is just a follow up to delete the temp file that #17012 forgot to do it. IMO, not need to add a comment for it.